### ISSUE TITLE
Anyone care to update this addon?

### ISSUE THREAD

**Issue by MHultman** [icanos/hassio-plejd#307]
Hi, 

Doesn't seem that we got the attention to get a new release of the addon over the last couple of months. 
What is going on?

I would love to see PR[#297](https://github.com/icanos/hassio-plejd/pull/297) in 0.12.0.

**Comment by irisityvictorhagelback** [comment]
Hi,
I'm interested in that as well (as the author). Have you tested the PR? 

As I write in the PR I'd like more people testing it before merging.

An in general it's an open source initiative, so please contribute!

/Victor

**Comment by SweVictor** [comment]
Hi,
I'm interested in that as well (as the author). Have you tested the PR?

As I write in the PR I'd like more people testing it before merging.

An in general it's an open source initiative, so please contribute!

/Victor

**Comment by tomoqv** [comment]
How do I try this out? I have massive problems getting plejd to work on my Intel NUC with built-in BT.

```
 Add-on version: 0.11.0
 You are running the latest version of this add-on.
parse error: Expected string key before ':' at line 1, column 4
[18:03:03] ERROR: Unknown HTTP error occured
 System:   (amd64 / generic-x86-64)
 Home Assistant Core: 2024.8.1
 Home Assistant Supervisor: 2024.08.0


```

**Comment by SweVictor** [comment]
> How do I try this out? I have massive problems getting plejd to work on my Intel NUC with built-in BT.
> 
> ```
>  Add-on version: 0.11.0
>  You are running the latest version of this add-on.
> parse error: Expected string key before ':' at line 1, column 4
> [18:03:03] ERROR: Unknown HTTP error occured
>  System:   (amd64 / generic-x86-64)
>  Home Assistant Core: 2024.8.1
>  Home Assistant Supervisor: 2024.08.0
> ```

The logs posted are known but do not affect anything. 

To install a PR, use the manual install method but download the code from the PR branch instead of the latest master

**Comment by tomoqv** [comment]
Sorry for being a bit of a noob with Github, by I fail to find the code download link on this page: https://github.com/icanos/hassio-plejd/pull/297

**Comment by MHultman** [comment]
The PR says it’s merged to develop branch so should work with that one. https://github.com/icanos/hassio-plejd/tree/develop

**Comment by tomoqv** [comment]
Now running 12.0-beta, but can't get it to work at all with my plejd devices.
[plejd-12.0-beta.log](https://github.com/user-attachments/files/16599750/plejd-12.0-beta.log)

**Comment by tomoqv** [comment]
Also, plejd devices show as switches instead of, as before, lamps in the HA gui.
![image](https://github.com/user-attachments/assets/366d0285-12ee-45e9-a269-1667af77ee12)

**Comment by SweVictor** [comment]
> Also, plejd devices show as switches instead of, as before, lamps in the HA gui.

How do they look in Plejd? What's the device type for the devices that are "wrong"?

From memory I think we follow the device type more correctly now and that you can use HA functionality (helpers?) to change device type in HA.

**Comment by tomoqv** [comment]
I have used HA helpers to turn them into lamps, but now they show as only switches in HA. Problem is, I haven't got the Plejd integration to work since the update to the beta version. The stable version I used before that, was nothing but stable, but worked sometimes. I haven't tried to roll back to the latest stable version, but I may do that as it at least worked occasionally. Sadly, the newer beta version seems to not work at all.

Please tell me which logs I need to provide, and I'll be glad to help out and test.

**Comment by SweVictor** [comment]
Well... As a first step, what do the logs say in the addon? Have you tried restarting the computer running HA?

If you can't figure out what's wrong them I would increase log level to verbose, pinpoint when things go wrong and post here. 

Look in the readme how to get full logs if what you see in the plugin window is not enough

**Comment by tomoqv** [comment]
I have power cycled my Intel NUC HASS machine and turned on verbose logging. Log file attached.
[plejd-12.0-beta.log](https://github.com/user-attachments/files/16715055/plejd-12.0-beta.log)
Hope this helps!
I am starting to suspect there is something wrong with the bluetooth adapter, but running bluetoothctl doesn't cleary indicate that for me. The fact that I previously has got Plejd working in HA by power cycling the NUC makes me wonder though... What do you think?

**Comment by MHultman** [comment]
From what I can see in your logs, the add-on is failing to start Bluetooth discovery and keeps throwing the error:

```
Failed to start discovery. Operation already in progress
DBusError: Operation already in progress
```
This is likely why you're stuck in a loop of failed reconnect attempts. It appears that the system can't begin discovery because another process might already be using the Bluetooth adapter.

The log itself suggests a possible solution:

Access the root console.
Run the following commands:
- bluetoothctl
- power off
- power on
- exit

After that, restart the Plejd add-on.

**Comment by tomoqv** [comment]
Yes, I tried that, but didn't look closely enough. This is what happens when I issue "power off":
![bild](https://github.com/user-attachments/assets/29f0a224-a6cc-4fa0-9643-b66903faec38)
It is turning the controller on again! I really don't know what's going on.

**Comment by MHultman** [comment]
It looks like your Bluetooth controller is automatically powering back on after you issue the power off command. This behavior could be caused by an automatic process or service that re-enables Bluetooth after it’s turned off.

Try some of this (Yes, I got if from GPT, but seems legit):

- **Check for Auto-Start Services:** There may be a service or process in your system that automatically restarts Bluetooth. Try checking if any services (e.g., bluetoothd) are set to restart automatically. You can temporarily stop the Bluetooth service with:

`sudo systemctl stop bluetooth`
After that, try running bluetoothctl again and power off the controller to see if it remains off.

- **Disable Automatic Power-On Features:** Some Bluetooth controllers have power management features that automatically turn them on after being powered off. If this is the case, you may need to look into disabling this feature through the Bluetooth configuration or your system's power management settings.

- **Look for Conflicting Add-ons:** As the log suggested earlier, it’s possible another add-on or process is scanning for Bluetooth devices and turning the controller back on. Make sure no other services or add-ons are trying to access the Bluetooth adapter during this time.

Try stopping the Bluetooth service as mentioned above and manually power it off again to see if the issue persists. Let us know if this changes anything!

**Comment by tomoqv** [comment]
I am also using this integration: https://www.home-assistant.io/integrations/bluetooth
But I have tried to disable it and repeat the above, and I am getting the same log errors and warnings.

**Comment by Wilbers007** [comment]
Hi all, I have managed to get the MQTT Mosquitto broker installed and also got the MQTT installed through HACS but when I try to add the Hassio-Plejd repository add on, it appears in the repositories but not in the add-ons page. What am I doing wrong? Thank you!
